src/Functions/FunctionsStringDistance.cpp

```diff
@@ -271,6 +272,222 @@ struct ByteEditDistanceImpl
     }
 };
 
+struct ByteDamerauLevenshteinDistanceImpl
+{
+    using ResultType = UInt64;
+    static constexpr size_t max_string_size = 1u << 16;
```

Comment:
Looking at the time complexity of the algorithm, I think the value is too optimistic. Please measure (using a release build) how long 1 million strings of varying lengths need to compute. I guess a second is acceptable but beyond that: no.